### PR TITLE
sql: adding eval method to drivers

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: schema/schemaspec/schemahcl/hcl.go [hunk 1]
```diff
@@ -38,8 +38,20 @@ type (
 	State struct {
 		config *Config
 	}
+	// Evaluator evaluates the Atlas DDL document with the given input and stores it in
+	// the provided object.
+	Evaluator interface {
+		Eval([]byte, interface{}, map[string]string) error
+	}
+	// EvalFunc is the function type that is implemented by the Eval method of the Evaluator interface.
```
Comment (masseelch) [inline schema/schemaspec/schemahcl/hcl.go:46]: Can we rephrase this to sth like "allows the use of ordinary functoin as Evaluator", because I am having trouble understanding that sentence :-D

File: sql/mysql/sqlspec.go [hunk 2]
```diff
@@ -19,10 +19,10 @@ type doc struct {
 	Schemas []*sqlspec.Schema `spec:"schema"`
 }
 
-// UnmarshalSpec unmarshals an Atlas DDL document using an unmarshaler into v.
-func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{}) error {
+// evalSpec unmarshals an Atlas DDL document using an unmarshaler into v.
```
Comment (masseelch) [inline sql/mysql/sqlspec.go:22]: update doc

File: sql/postgres/sqlspec.go [hunk 3]
```diff
@@ -33,10 +33,10 @@ func init() {
 	schemaspec.Register("enum", &Enum{})
 }
 
-// UnmarshalSpec unmarshals an Atlas DDL document using an unmarshaler into v.
-func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{}) error {
+// evalSpec evaluates an Atlas DDL document using an unmarshaler into v using the input.
```
Comment (masseelch) [inline sql/postgres/sqlspec.go:36]: The comments in the different driver on this method differ. Here you say `...v using...` and in SQLite it says `...v by using...`

**Review by masseelch — APPROVED** [review]

**Review by a8m — APPROVED** [review]

File: schema/schemaspec/schemahcl/hcl.go [hunk 4]
```diff
@@ -43,7 +43,7 @@ type (
 	Evaluator interface {
 		Eval([]byte, interface{}, map[string]string) error
 	}
-	// EvalFunc is the function type that is implemented by the Eval method of the Evaluator interface.
+	// EvalFunc is an adapter that allows the use of an ordinar function as an Evaluator.
```
Comment (masseelch) [inline schema/schemaspec/schemahcl/hcl.go:46]:
```suggestion
	// EvalFunc is an adapter that allows the use of an ordinary function as an Evaluator.
```

**Review by masseelch — APPROVED** [review]
